python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -98,63 +92,45 @@ def _read_paths(
 
         dataset_kwargs = dataset_kwargs or {}
         dataset_kwargs["partitioning"] = partitioning or "hive"
-        with ExitStack() as stack:
-            # Non-local filesystem handling
-            paths_or_fobs = paths
-            if not _is_local_filesystem(fs):
-                paths_or_fobs = _open_remote_files(
-                    paths_or_fobs,
-                    fs,
-                    context_stack=stack,
-                    **_default_open_file_options(
-                        open_file_options, columns, row_groups
-                    ),
-                )
```

Comment:
IIUC it's replacing this with `_fsspec_data_transfer` that's the relevant change, right?

Reply:
Yes exactly. Instead of trying to open the files in a way that pre-fetches remote data (and represents it as a `NativeFile`), we are not just looping over the remote files and transferring the byte ranges right here (using `_fsspec_data_transfer`).